packages/bitcoin-cash-rest-api-provider/lib/BitcoinCashRestApiProvider.ts

```diff
@@ -0,0 +1,197 @@
+    })
+
+    this._network = network
+    this._numberOfBlockConfirmation = numberOfBlockConfirmation
+    this._defaultFeePerByte = defaultFeePerByte
+    this._usedAddressCache = {}
+    this.bchjs = new BCHJS({ restURL: url })
+  }
+
+  async getFeePerByte(numberOfBlocks = this._numberOfBlockConfirmation) {
+    numberOfBlocks as any
+    // BCH fees are fairly stable
+    return this._defaultFeePerByte
+  }
+
+  async getMinRelayFee() {
+    return 1
+  }
+
+  async getBalance(_addresses: (string | Address)[]) {
+    const addresses = _addresses.map(addressToString)
+    const response = await this.bchjs.Electrumx.balance(addresses)['balances']
+    const sum = new BigNumber(0)
+
+    response.reduce((acc: BigNumber, balance: any) => acc.plus(balance['confirmed']).plus(balance['unconfirmed']), sum)
+
+    return sum
+  }
+
+  async _getUnspentTransactions(address: string): Promise<bitcoin.UTXO[]> {
+    const data = await this.bchjs.Electrumx.utxo(address)
+
+    return data['utxos'].map((utxo: any) => ({
+      address,
+      txid: utxo['tx_hash'],
+      vout: utxo['tx_pos'],
+      value: utxo['value'],
+      blockHeight: utxo['height']
+    }))
+  }
+
+  async getUnspentTransactions(_addresses: (Address | string)[]): Promise<bitcoin.UTXO[]> {
+    const addresses = _addresses.map(addressToString)
+    const utxoSets = await Promise.all(addresses.map((addr) => this._getUnspentTransactions(addr)))
+    const utxos = flatten(utxoSets)
+    return utxos
+  }
+
+  async _getAddressTransactionCount(address: string) {
+    const transactions = await this.bchjs.Electrumx.transactions(address)
+    return transactions['transactions'].length
+  }
+
+  async getAddressTransactionCounts(_addresses: (Address | string)[]) {
+    const addresses = _addresses.map(addressToString)
+    const transactionCountsArray = await Promise.all(
+      addresses.map(async (addr) => {
+        const txCount = await this._getAddressTransactionCount(addr)
+        return { [addr]: txCount }
+      })
+    )
+    const transactionCounts = Object.assign({}, ...transactionCountsArray)
+    return transactionCounts
+  }
+
+  async getTransactionHex(transactionHash: string): Promise<string> {
+    return await this.bchjs.RawTransactions.getRawTransaction(transactionHash)
+  }
+
+  async getTransaction(transactionHash: string) {
+    const currentHeight = await this.getBlockHeight()
+
+    let rawTx = await this.bchjs.RawTransactions.getRawTransaction([transactionHash], true)
+    rawTx = rawTx[0]
+    const status: explorer.TxStatus = { confirmed: rawTx['confirmations'] > 0 }
+    if (status.confirmed) {
+      status.block_height = currentHeight - rawTx['confirmations'] + 1
+      status.block_hash = await this.getBlockHash(status.block_height)
+    }
+
+    const tx: explorer.Transaction = {
+      txid: rawTx['txid'],
+      version: rawTx['version'],
+      locktime: rawTx['locktime'],
+      vin: rawTx['vin'],
+      fee: 0, // TODO
```

Comment:
This is not a "will do" but rather a "don't care" that I justified in the last section of [here](https://github.com/liquality/chainabstractionlayer/pull/405#issue-630515718)